src/OpenTelemetry/Extensions/OpenTelemetryExtensions.cs

```diff
@@ -0,0 +1,117 @@
+
+                            var headers = Environment.GetEnvironmentVariable("OTEL_EXPORTER_OTLP_TRACES_HEADERS")
+                                ?? Environment.GetEnvironmentVariable("OTEL_EXPORTER_OTLP_HEADERS");
+                            options.Headers = headers;
+                        });
+                        break;
+                    case "zipkin":
+                        tracerProviderBuilder.AddZipkinExporter(options =>
+                        {
+                            options.Endpoint = new Uri(openTelemetryConfig.ExportToolEndpoint);
+                        });
+                        break;
+                    default:
+                        tracerProviderBuilder.AddConsoleExporter();
+                        break;
+                }
+            });
+        }
+
+        public static void AddOpenTelemetry(ConnectionMultiplexer connectionMultiplexer, OpenTelemetryConfig openTelemetryConfig)
+        {            
+
+            if (openTelemetryConfig == null || openTelemetryConfig.ExportType == null)
+            {
+                return;
+            }
+
+            var tracerProviderBuilder = Sdk.CreateTracerProviderBuilder();
+
+            // Configure resource
+            tracerProviderBuilder
+                .SetResourceBuilder(ResourceBuilder.CreateDefault().AddService(openTelemetryConfig.ServiceName));
+
+            // Configure instrumentation
+            tracerProviderBuilder
+                .AddAspNetCoreInstrumentation()
+                .AddHttpClientInstrumentation()
+                .AddRedisInstrumentation(connectionMultiplexer);
+
+            // Configure exporter
+            switch (openTelemetryConfig.ExportType)
+            {
+                case "jaeger":
+                    tracerProviderBuilder.AddJaegerExporter(options =>
+                    {
+                        options.AgentHost = openTelemetryConfig.ExportToolEndpoint;
+                    });
+                    break;
+                case "otlp":
+                    tracerProviderBuilder.AddOtlpExporter(options =>
+                    {
+                        options.Endpoint = new Uri(openTelemetryConfig.ExportToolEndpoint);
+
+                        var headers = Environment.GetEnvironmentVariable("OTEL_EXPORTER_OTLP_TRACES_HEADERS")
+                            ?? Environment.GetEnvironmentVariable("OTEL_EXPORTER_OTLP_HEADERS");
+                        options.Headers = headers;
+                    });
+                    break;
+                case "zipkin":
+                    tracerProviderBuilder.AddZipkinExporter(options =>
+                    {
+                        options.Endpoint = new Uri(openTelemetryConfig.ExportToolEndpoint);
+                    });
+                    break;
+                default:
+                    tracerProviderBuilder.AddConsoleExporter();
+                    break;
+            }
+
+            tracerProviderBuilder.Build();
```

Comment:
If the provider being built here is not stored, then it cannot be disposed.

Reply:
Thank you @cijothomas, for the review and sorry for the delay in updating the changes. 
Yes, updated the file `src/OpenTelemetry/Extensions/OpenTelemetryExtensions.cs` and used the `tracerProviderBuilder` as local parameter.